### PR TITLE
fix component viewer bug

### PR DESCRIPTION
**Pull request by mbarekh**

### CODE REVIEW

File: viewer/src/componentModel.ts [hunk 1]
```diff
@@ -254,7 +259,15 @@ export class ComponentModel {
                 "text": text
             });
         }
-        const triggerableLinksJson = modelJson.ComponentData.TransversalLinks[0].TransversalTransitionData || [];
+
+        // let triggerableLinksJson = modelJson.ComponentData.TransversalLinks === undefined ? [] : modelJson.ComponentData.TransversalLinks[0].TransversalTransitionData;
```
Comment (nicolaserny) [inline viewer/src/componentModel.ts]: Remove comment

File: viewer/src/componentModel.ts [hunk 2]
```diff
@@ -254,7 +259,15 @@ export class ComponentModel {
...
+        let triggerableLinksJson;
```
Comment (nicolaserny) [inline viewer/src/componentModel.ts]: your code seems to be too complicated (why checking undefined twice ?)
ex:
const triggerableLinksJson = modelJson.ComponentData.TransversalLinks && modelJson.ComponentData.TransversalLinks.length > 0 ? odelJson.ComponentData.TransversalLinks[0].TransversalTransitionData : [];

File: viewer/src/componentModel.ts [hunk 3]
```diff
@@ -300,7 +313,12 @@ export class ComponentModel {
         }
         // transition pattern state
         const transitionPatternStates = {};
-        let transitionPatternStateDataJson = modelJson.ComponentData.TransitionPatternStates[0].TransitionPatternStateData;
+        let transitionPatternStateDataJson = modelJson.ComponentData.TransitionPatternStates === undefined ? [] : modelJson.ComponentData.TransitionPatternStates[0].TransitionPatternStateData;
```
Comment (nicolaserny) [inline viewer/src/componentModel.ts]: same comment as triggerableLinksJson

**Review by nicolaserny — CHANGES_REQUESTED** [review]
Add unit tests related to the bug

File: viewer/src/componentModel.ts (hunk 1 above)
Comment (mbarekh) [reply viewer/src/componentModel.ts]: done

File: viewer/src/componentModel.ts (hunk 2 above)
Comment (mbarekh) [reply viewer/src/componentModel.ts]: done

File: viewer/src/componentModel.ts (hunk 3 above)
Comment (mbarekh) [reply viewer/src/componentModel.ts]: done

**Comment by mbarekh** [comment]
Added a unit test for the bug

**Review by nicolaserny — APPROVED** [review]

**Comment by nicolaserny** [comment]
resolves #108

**Pull request merged**